proxy/src/control/discovery.rs

```diff
@@ -187,10 +219,17 @@ impl Background {
 
 // ==== impl DiscoveryWork =====
 
-impl<T> DiscoveryWork<T>
+impl<E, T, S> DiscoveryWork<T, S>
 where
     T: HttpService<RequestBody = BoxBody>,
+    // T::Future: fmt::Debug,
     T::Error: fmt::Debug,
+    E: fmt::Debug,
+    UpdateRsp<T::Future>: Future<Item=grpc::Response<S>, Error=E>,
+    S: Stream<Item=PbUpdate, Error=E>,
+    <UpdateRsp<T::Future> as Future>::Error: fmt::Debug,
+    UpdateRx<UpdateRsp<T::Future>, S>: Stream<Item=PbUpdate>,
+    <UpdateRx<UpdateRsp<T::Future>, S> as Stream>::Error: fmt::Debug,
```

Comment:
I haven't really figured out why all of these where bounds are required. Is there anyway to trim this down?

Reply:
@carllerche this is my best attempt at reducing the number of bounds; I can try to see if some of them can be removed, but it used to look [much worse](https://github.com/carllerche/conduit/pull/2/commits/5138f0392a391978484d4d3d7fb638f717db8232?diff=unified#diff-3d7563188396c0392342b192778deb15R209). I'll see if it can be reduced further, but I doubt it.